source/CommonJobs/CommonJobs.Application.EvalForm/Commands/UpdateCalificationsCommand.cs

```diff
@@ -0,0 +1,126 @@
+﻿using CommonJobs.Application.EvalForm.DTOs;
+using CommonJobs.Domain.Evaluations;
+using CommonJobs.Infrastructure.RavenDb;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace CommonJobs.Application.EvalForm.Commands
+{
+    public class UpdateCalificationsCommand : Command
+    {
+        private const string COMPANY = "_company";
+
+        private UpdateEvaluationDTO _updateEvaluation;
+
+        private string _loggedUser;
+
+        public UpdateCalificationsCommand(UpdateEvaluationDTO updateEvaluation, string loggedUser)
+        {
+            _updateEvaluation = updateEvaluation;
+
+            _loggedUser = loggedUser;
+        }
+
+        public override void Execute()
+        {
+            bool updateEvaluationComments = false;
+            bool updateEvaluationProject = false;
+
+            var storedEvaluation = RavenSession.Load<EmployeeEvaluation>(_updateEvaluation.EvaluationId);
+            if (storedEvaluation == null)
+            {
+                throw new ApplicationException(string.Format("Error: Evaluación {0} inexistente", _updateEvaluation.EvaluationId));
+            }
+
+            foreach (var calification in _updateEvaluation.Califications)
+            {
+                var storedCalification = RavenSession.Load<EvaluationCalification>(calification.CalificationId);
+                if (storedCalification == null)
+                {
+                    throw new ApplicationException(string.Format("Error: Calificación {0} inexistente para evaluación {1}", calification.CalificationId, _updateEvaluation.EvaluationId));
+                }
+
+                // Checks if the user trying to update the califications can actually do it
+                if (CanUpdate(_loggedUser, storedEvaluation, storedCalification))
+                {
+                    UpdateCalification(calification, storedCalification, _updateEvaluation.Finished);
+
+                    // If it's the responsible's or the company's calification, then the evaluation project should be updated [ TODO: Improvement - Check if the project has changed before updating it ]
+                    updateEvaluationProject = (storedCalification.Owner == CalificationType.Responsible || storedCalification.Owner == CalificationType.Company);
```

Comment:
I am not sure to understand, the flag is being evaluated inside a foreach (see [line 37](https://github.com/CommonJobs/CommonJobs/pull/387/files#diff-e00df1392631659f1018fc0f93e90667R37)), but it is being used outside (see [line 61](https://github.com/CommonJobs/CommonJobs/pull/387/files#diff-e00df1392631659f1018fc0f93e90667R61)). Is it fine?